static/js/contact_form.js

```diff
@@ -1,71 +1,74 @@
-  $scope._errors = {};
+    $scope.user = { gender:'male', t_shirt_size:'m', 'english_knowledge':'elementary', 'experience_level':'newbie' };
+    $scope._errors = {};
 
-        if ($window.localStorage) {
-            var user = $window.localStorage.getItem('user');
-            if (user) {
-//		  $http.get('api/participants/'+user,{params:{uid:user}}).then(function(r){ $scope.user = r.data});
-                try {
-                    $scope.user = JSON.parse(user);
-                } catch(err) {};
-            }
+    if ($window.localStorage) {
+        var user = $window.localStorage.getItem('user');
+        if (user) {
+            try {
+                $scope.user = JSON.parse(user);
+            } catch(err) {};
         }
+    }
 
-  if (GoogleUser != null) {
-      angular.extend($scope.user, {
-          name: GoogleUser.given_name,
-          surname: GoogleUser.family_name,
-          gender: GoogleUser.gender,
-          email: GoogleUser.email,
-          gplus: GoogleUser.link
-      });
+    if (GoogleUser != null) {
+        angular.extend($scope.user, {
+            name: GoogleUser.given_name,
+            surname: GoogleUser.family_name,
+            gender: GoogleUser.gender,
+            email: GoogleUser.email,
+            gplus: GoogleUser.link
+        });
 
-      $scope.picture = GoogleUser.picture;
-      $scope.fromGoogle = true;
-  } else $scope.fromGoogle = false;
+        $scope.picture = GoogleUser.picture;
+        $scope.fromGoogle = true;
+    } else $scope.fromGoogle = false;
 
     var parts = $window.location.pathname.split('/');
     if (parts[0] == '') parts.shift();
     if (parts[0] == 'events') parts.shift();
     var event = parts[0];
 
+    $scope.submit = function() {
+        if (!$scope.contactForm.$valid) {
+            console.log("form is not valid");
+            return;
+        }
 
-  $scope.submit = function() {
-    if (!$scope.contactForm.$valid) {
-	console.log("form is not valid");
-		return;
-   }
-
-    console.log("Saving");
-    $scope.showOk = false;
-    $scope.saving = true;
-    $scope.showError = false;
-    var savedCb = function(r) {
+        console.log("Saving");
+        $scope.showOk = false;
+        $scope.saving = true;
+        $scope.showError = false;
+        var savedCb = function(r) {
 
-       $scope.saving = false;
+           $scope.saving = false;
 
-       if ( r.status!=200 || r.data.code) {
-	        $scope.showError = true;
+           if ( r.status!=200 || r.data.code) {
+                $scope.showError = true;
                 $scope._errors = r.data.hasOwnProperty('errors') ? r.data.errors : {};
-       		console.log("error:", r);
-
-       } else {
-	        $scope.showOk = true;
+                // Do something with incoming dicts
+                for (key in $scope._errors) {
+                    $scope._errors[key] = $scope._errors[key].join(' ');
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["key" is not defined.](https://www.codacy.com/app/webknjaz/gdg-org-ua/file/2612254393/issues/source?bid=3283927&fileBranchId=3285896#l52)